runtime/onert/frontend/base_loader/include/base_loader.h

```diff
@@ -243,6 +245,31 @@ void BaseLoader<LoaderDomain>::BaseLoader::loadFromBuffer(uint8_t *buffer, size_
   loadModel();
 }
 
+template <typename LoaderDomain>
+std::unique_ptr<ir::Data>
+BaseLoader<LoaderDomain>::BaseLoader::loadMetadata(const uint32_t buffer_idx)
+{
+  assert(_domain_model != nullptr);
+  const auto *data = _domain_model->buffers()->Get(buffer_idx)->data();
+  if (_fd == -1) // Model is from memory
+  {
+    return std::make_unique<ir::ExternalData>(data->data(), data->size());
+  }
+  else // Model is loaded(mmap'd) from a file
+  {
+    size_t data_size = data->size();
+    ptrdiff_t offset_start = data->data() - _base;
+    ptrdiff_t offset_end = offset_start + data_size;
+
+    ptrdiff_t page_start = (offset_start / _pagesize) * _pagesize;
+    size_t mapping_size = offset_end - page_start;
+
+    // Since metadata is not access often in inference/training time, always use mmaped-data
+    // Ref : https://github.com/Samsung/ONE/issues/3961#issuecomment-681750231
+    return std::make_unique<ir::MMapedData>(_fd, page_start, mapping_size, offset_start, data_size);
```

Comment:
`_fd` must outlive returned `std::unique_ptr<ir::Data>`. Currently @zetwhite is going to convert `ir::Data` to `TrainingInfo` in model loading phase. Thus, it will be safe at this moment.

Reply:
Ah, I also worried about this part. 

As I searched, there is no need to keep `_fd` open after mmaped.

--------------------------

I refered this : https://stackoverflow.com/questions/17490033/do-i-need-to-keep-a-file-open-after-calling-mmap-on-it  


* From Posix manpage [ref](https://pubs.opengroup.org/onlinepubs/7908799/xsh/mmap.html#:~:text=The%20mmap()%20function%20adds%20an%20extra%20reference%20to%20the%20file%20associated%20with%20the%20file%20descriptor%20fildes%20which%20is%20not%20removed%20by%20a%20subsequent%20close()%20on%20that%20file%20descriptor.%20This%20reference%20is%20removed%20when%20there%20are%20no%20more%20mappings%20to%20the%20file.) 

> The mmap() function adds an extra reference to the file associated with the file descriptor fildes which is not removed by a subsequent close() on that file descriptor. This reference is removed when there are no more mappings to the file. 

* From Linux manpage [ref](https://linux.die.net/man/2/mmap#:~:text=On%20the%20other%20hand%2C%20closing%20the%20file%20descriptor%20does%20not%20unmap%20the%20region.) 

> On the other hand, closing the file descriptor does not unmap the region.